site/_data/postsData.js

```diff
@@ -0,0 +1,82 @@
+const yaml = require('js-yaml');
+const fs = require('fs');
+const glob = require('glob');
+const path = require('path');
+const matter = require('gray-matter');
+
+const {defaultLocale} = require('./site.json');
+const PLACEHOLDER_IMG =
+  'image/tcFciHGuF3MxnTr1y5ue01OGLBn2/PFaMfvDZoPorronbpdU8.svg';
+
+/** @type AuthorsData */
+const authorsData = require('./authorsData.json');
+const authorsYaml =
+  /** @type I18nAuthors */ (
+    /** @type TODO */ yaml.load(
+      fs.readFileSync(path.join(__dirname, 'i18n', 'authors.yaml'), 'utf-8')
+    )
+  ) || {};
+
+/** @type PostsData */
+const postsData = {
+  authors: {},
+};
+
+/**
+ * @param {string} authorKey
+ * @param {FrontMatterData} frontMatter
+ */
+function addAuthorsPost(authorKey, frontMatter) {
+  if (authorKey in postsData.authors) {
+    postsData.authors[authorKey].elements.push(frontMatter);
+  } else {
+    postsData.authors[authorKey] = {
+      image: PLACEHOLDER_IMG,
+      ...authorsData[authorKey],
+      description: `i18n.authors.${authorKey}.description`,
+      elements: [frontMatter],
+      key: authorKey,
+      title: `i18n.authors.${authorKey}.title`,
+      url: `/authors/${authorKey}/`,
+    };
+  }
+}
+
+const files = glob.sync('./site/[a-z][a-z]/{**/*,*}.md', {});
+
+/** @type {FrontMatterData[]} */
+const fileContents = files
+  .map(file => {
+    const filePath = path.join(process.cwd(), file);
+    const fileData = fs.readFileSync(filePath, 'utf8');
+    const data = /** @type FrontMatterData */ (
+      /** @type TODO */ matter(fileData).data
+    );
+
+    const localeGroups = /\/site\/(?<locale>[a-z]{2})\/.*/.exec(file)?.groups;
```

Comment:
These regexp are kinda gross. I wonder if we could:

- change `glob` to make its results relative to `./site` 
- pop the left-side string and check it's two-characters long
- use the rest of the result as the url

Happy to chat more though.

Reply:
I think/thought I was already getting relative paths from my glob (such as `./site/en/tags/index.md`). Ohhh, I think you mean so that all paths returned would look more like `./en/tags/index.md`? I'm genuinely unsure how to do that. Though I could make the regex group into one object:

```js
const regexGroups = /\/site\/(?<locale>[a-z]{2})(?<url>\/.*\/).*\.md$/.exec(
      file
    )?.groups;
```

Without being able to do the relative bit I'm not sure how I could effectively or accurately do the rest... 🥲